src/elements/List/lessToJS.js

```diff
@@ -0,0 +1,85 @@
+const fs = require('fs')
+const path = require('path')
+const _ = require('lodash')
+
+const wrapLessVariablesFile = (type, name, content) => `// Converted from:
+// https://github.com/Semantic-Org/Semantic-UI-LESS/blob/master/themes/default/${type}/${_.lowerCase(name)}.variables
+module.exports = (siteVars) => {
+  const vars = {}
+
+${content}
+  return vars
+}
+`
+
+
+exports.varsFile = (type, name, filepath) => {
+  const absPath = path.resolve(process.cwd(), filepath)
+
+  const content = fs.readFileSync(absPath, 'utf8').toString()
+    .replace(/\s*:\s*/g, ' = ')                       // colon to equal sign
+
+    //
+    // PRE FORMATTING
+    //
+    .replace(/\/\*\*+[\S\s]+?\*\*\/\n+/, '')          // remove component name header
+    .replace(/\s+;/gm, ';')                           // semis to end of line
+
+    //
+    // VARIABLE REFERENCES
+    //
+    .replace(/^@/gm, 'vars.')                         // make variable defs members of vars
+
+    // Variables used in the value position can be defined locally or globally.
+    // Determine if a variable has been defined in this file prior to its use.
+    // Properly reference the variable from vars (local) or siteVars (global).
+    .replace(/@(\w+)/gm, (match, g1, offset, string) => {
+      const precedingContent = string.substr(0, offset)
+      const localVar = `vars.${g1} =`
+      const localVarRegExp = new RegExp(localVar, 'g')
+      const isLocalVar = localVarRegExp.test(precedingContent)
+
+      const prefix = isLocalVar ? 'vars' : 'siteVars'
+
+      return `\${${prefix}.${g1}}`
+    })            // BAD assume other vars are siteVars
+
+    //
+    // VALUES
+    //
+    // Make all values template strings, since all vars in values are already in template form ${}.
+    // Then, convert any templates without interpolation to regular strings.
+    // Then, convert any templates with a single interpolated value to a regular variable reference.
+    // Then, fix any doubly single quoted strings
+    .replace(/= ([\S\s]+?);/gm, '= `$1`;')       // back tick ALL values
+    .replace(/= `([^$]+?)`;/gm, "= '$1';")       // single quote non tpls
+    .replace(/`\${(siteVars\.\w+)}`/gm, '$1')    // remove unnecessary back ticks
+    .replace(/''([\S\s]+?)''/g, "'$1'")          // fix nested single quotes
+
+    // PARENS
+    .replace(/(\W)\((.*)\)/g, '$1calc($2)')             // LESS parens to CSS calc()
+
+    // POST FORMATTING
+    .replace(/^(.+)/gm, '  $1')                       // indent lines
+    .replace(/;/g, '')                                // remove semis
+
+  return wrapLessVariablesFile(type, name, content)
+}
```

Comment:
⏫ This was the part I was wondering how we would handle. Love it.

I still have not had a chance to dig into this because of a release we need to ship this week... and I am heading out of town for a wedding on Thursday. I'll take a look on the plane though for sure.

Reply:
It isn't pretty, but it actually is working quite nicely so far!  I was testing this out on the Label component by manually pulling things over, now I'm automating that and testing against the List component.  Next, I'll try to do this with the site variables and the entire LESS build.

Once we have this pulled over once, we won't have to do it again.  I am quite certain that we'll outpace SUI core CSS/LESS in features and fixes due to the size and speed of our community.  That means we won't constantly update from them anymore but move forward instead.

Reply:
> Once we have this pulled over once, we won't have to do it again.

I was actually wondering about your thoughts on that. I guess technically we could just reference the npm package for those definitions. We could have a script that would just convert them all again once SUI updates and we update our package.

I guess though that if we move forward at a faster pace we might potentially have significant conflicts to resolve.